### PR TITLE
Travis: running more tests for cache component

### PR DESCRIPTION
**Pull request by marc-mabe**
- enabled `Apc` storage adapter tests using last available `apcu` extension
- enabled `Memcache` storage adapter tests using pre-installed but not enabled `memcache` extension
- enabled `Memcached` storage adapter tests using pre-installed but not enabled `memcached` extension
  - start memcached on `before_script` 
- enabled `Redis` storage adapter tests using pre-installed but not enabled `redis` extension
  - start redis-server on `before_script` 
- enabled all other storage adapter tests even if they are not available.
  - the tests will be skipped automatically but this enables testing of the availability check + expected exception
- removed constant for SQLite because the is no Sqlite storage adapter yet

There are some tests failing currently but these should be solved by #7065 and #7043

@Ocramius ping

### CODE REVIEW

File: tests/TestConfiguration.php.dist [hunk 1]
```diff
@@ -71,7 +71,6 @@ defined('TESTS_ZEND_BARCODE_PDF_SUPPORT') || define('TESTS_ZEND_BARCODE_PDF_SUPP
  * Zend\Cache\Storage\Adapter
  */
 defined('TESTS_ZEND_CACHE_APC_ENABLED') || define('TESTS_ZEND_CACHE_APC_ENABLED', false);
-defined('TESTS_ZEND_CACHE_SQLITE_ENABLED') || define('TESTS_ZEND_CACHE_SQLITE_ENABLED', false);
```
Comment (weierophinney) [inline tests/TestConfiguration.php.dist]: Why are you removing this one?

File: tests/TestConfiguration.php.travis [hunk 2]
```diff
@@ -42,6 +42,21 @@ defined('TESTS_ZEND_HTTP_CLIENT_ONLINE') || define('TESTS_ZEND_HTTP_CLIENT_ONLIN
  * Zend\Cache\Storage\Adapter
  */
 defined('TESTS_ZEND_CACHE_APC_ENABLED') || define('TESTS_ZEND_CACHE_APC_ENABLED', true);
-defined('TESTS_ZEND_CACHE_SQLITE_ENABLED') || define('TESTS_ZEND_CACHE_SQLITE_ENABLED', true);
```
Comment (weierophinney) [inline tests/TestConfiguration.php.travis]: Same question here...

File: tests/TestConfiguration.php.dist (hunk 1 above)
Comment (marc-mabe) [inline tests/TestConfiguration.php.dist]: There is no sqlite adapter - this is a relict of ZF1

Comment (weierophinney) [inline tests/TestConfiguration.php.dist]: Okay -- thanks for the clarification!

**Comment by weierophinney** [comment]
Merged to develop for release with 2.4.

**Pull request closed without merging**